### ISSUE TITLE
Performance issue

### ISSUE THREAD

**Issue by hyanmandian** [springload/react-accessible-accordion#110]
Hello :D I don't know why, but I have a performance issue with version 2.4.0, I think that the changes made from 2.3.1 to 2.4.2 make this happen. I don't have time to help you to found it right now, but I would like to warn.

I suppose that the lifecycles refactoring that you did make this issue (maybe why-did-you-update lib can help you to debug)

Thanks!

**Comment by ryami333** [comment]
Hi @hyanmandian, could you describe what you mean by a 'performance issue'? Have you actually verified that memory usage is increasing? Extra renders are happening? Would you please be able to send us a Codesandbox to demonstrate how you're using the library?

**Comment by hyanmandian** [comment]
Sure, good idea :D

Version: 2.3.1 -> https://codesandbox.io/s/5z55m8wk3k
Version: 2.4.2 -> https://codesandbox.io/s/3r30opwlv1

**Comment by hyanmandian** [comment]
If you use "why-did-you-update" library, you will see a lot of logs about Provider and Consumer components

**Comment by ryami333** [comment]
OK, thanks for the details. Will have a bit of dig!

**Comment by hyanmandian** [comment]
Maybe the issue is in this file -> https://github.com/springload/react-accessible-accordion/blob/master/src/AccordionItem/accordion-item-wrapper.js

**Comment by hyanmandian** [comment]
Sorry, miss click...

So, I think that this issue happen in this file -> https://github.com/springload/react-accessible-accordion/blob/master/src/AccordionItem/accordion-item-wrapper.js

Maybe because you create a Provider and Subscribe for each item

**Comment by hyanmandian** [comment]
I have an idea to solve that. I will send you a PR with this idea, just sec.

**Comment by hyanmandian** [comment]
I tried in here, but I think that I need to study how use TS first (I don't have any exp with TS)... But basically the idea is move the accordion-item-wrapper.js Provider and Subscribe components to accordion.js and pass those props to childrens using React.cloneElement.

Something like that:

```js
// accordion.js
render() {
        const { accordion, children, ...rest } = this.props;

        return (
            <Provider inject={[this.itemContainer]}>
                <Subscribe to={[AccordionContainer, ItemContainer]}>
                    {(accordionStore, itemStore) => {
                        const { uuid } = itemStore.state;
                        return (
                            <div role={accordion ? 'tablist' : null} {...rest}>
                                {React.Children.map(children, (item) => React.cloneElement(item, {
                                    accordionStore,
                                    itemStore,
                                }))}
                            </div>
                        );
                    }}
                </Subscribe>
            </Provider>
        );
    }
```

**Comment by ryami333** [comment]
@hyanmandian, our apologies for the delay - but I think I've resolved this. I've published a pre-release `2.4.4-beta.1` to check this out with your codesandbox and everything looks better to me now. Could you please check too? https://codesandbox.io/s/jz80xw6o29

**Comment by hyanmandian** [comment]
@ryami333 Sorry for the delay :(I tested the last version 2.4.4 and the problem is the same -> https://codesandbox.io/s/z60nz3nnv4
